mod_reforged/hooks/skills/perks/perk_anticipation.nut

```diff
@@ -1,119 +1,101 @@
 ::Reforged.HooksMod.hook("scripts/skills/perks/perk_anticipation", function(q) {
-	// Public
-	q.m.UsesMax <- 2;
-	q.m.DamageReductionPerTile <- 10.0;
-	q.m.DamageReductionBase <- 0.0;
+	// Config
+	q.m.BaseChance = 0;
+	q.m.ChancePerTile <- 10;
+	q.m.MaxChance = 70;
+	q.m.MinDamageReceivedTotalMult = 0.7;
 
 	// Private
-	q.m.UsesRemaining <- 2;
-
-	// Temp
-	q.m.TempDamageReduction <- 0.0;		// This is used to keep track of the current damage reduction so that it can be later displayed in the combat log
-	q.m.IsAboutToConsumeUse <- false;
+	// These two fields are used to ensure that the damage reduction is the same in all instances of damage from a single attack from an attacker
+	q.m.SkillCount <- 0;
+	q.m.LastAttacker <- null;
+	// This is used to keep track of the current damage reduction so that it can be later displayed in the combat log
+	q.m.TempDamageReceivedTotalMult <- 1.0;
 
 	q.create = @(__original) function()
 	{
 		__original();
-		this.m.Description = "Use your fast senses to anticipate attacks against you, taking reduced damage from the first few attacks each battle.";
+		this.m.Description = "Use your fast senses to anticipate attacks against you, taking reduced damage.";
 		this.m.IconMini = "rf_anticipation_mini";
 
 		this.addType(::Const.SkillType.StatusEffect);	// We now want this effect to show up on entities
 	}
 
 	q.isHidden <- function()
 	{
-		return (this.isEnabled() == false);
-	}
-
-	q.onCombatStarted <- function()
-	{
-		this.m.UsesRemaining = this.m.UsesMax;
-	}
-
-	q.onCombatFinished <- function()
-	{
-		this.m.UsesRemaining = this.m.UsesMax;	// So that for the purposes of the tooltip everything looks good
+		return this.getChance() == 0;
 	}
 
 	q.getTooltip <- function()
 	{
 		local ret = this.skill.getTooltip();
 
-		local damageReduction = this.calculateDamageReduction();
-		ret.extend([
-			{
-				id = 10,
-				type = "text",
-				icon = "ui/icons/melee_defense.png",
-				text = ::MSU.Text.colorNegative(damageReduction + "%") + " reduced damage received from all attacks"
-			},
-			{
+		local chance = this.getChance();
+		local mult = this.getDamageReceivedTotalMult();
+		ret.push({
+			id = 10,
+			type = "text",
+			icon = "ui/icons/melee_defense.png",
+			text = ::MSU.Text.colorPositive(chance + "%") + " chance to receive " + ::MSU.Text.colorizeMult(mult, {InvertColor = true}) + " reduced damage from attacks"
```

Comment:
You can use `colorizeMultWithText` here and then remove the work `reduced` from the text at the end